lib/fog/openstack/requests/network/update_subnet.rb

```diff
@@ -24,11 +24,12 @@ class Mock
         def update_subnet(subnet_id, options = {})
           response = Excon::Response.new
           if subnet = list_subnets.body['subnets'].find { |_| _['id'] == subnet_id }
-            subnet['name']            = options[:name]
```

Comment:
hm, seems like a pretty bad way to update

since e.g. when you want to change a name, you will just send :name => 'something', so the check should be the same as in create (different fields, since some can't be updated)

But again, not your fault, it's not scope of this PR.